src/pages/BuyDomain/BuyDomain.tsx

```diff
@@ -0,0 +1,70 @@
+import type { FC } from 'react';
+
+import { useState } from 'react';
+import {
+	IconZero,
+	DomainSearch,
+	BuyDomainModal,
+	ConnectWallet,
+} from '../../components';
+import { useWeb3 } from '../../hooks';
+import styles from './BuyDomain.module.scss';
+
+export const BuyDomain: FC = () => {
+	const { account } = useWeb3();
+
+	const [buyDomainModal, setBuyDomainModal] = useState<{
+		isOpen;
+		domainName;
+	}>({
+		isOpen: false,
+		domainName: '',
+	});
```

Comment:
You can remove `isOpen` from this state object, and render conditionally based on `domainName === undefined`. I would also change the name of the state object to reflect the change in logic, e.g. `const [ domainToPurchase, setDomainToPurchase ] = useState<string | undefined>()` or something similar.

Reply:
Updated.